plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/BuildPath.java

```diff
@@ -89,11 +89,19 @@ public static void addMavenLibraries(IProject project, List<Library> libraries,
       IJavaProject javaProject, List<Library> libraries, IProgressMonitor monitor)
           throws JavaModelException, CoreException {
     
+    return prepareLibraries(javaProject, libraries, monitor, true);
+  }
+
+  private static IClasspathEntry[] prepareLibraries(IJavaProject javaProject,
+      List<Library> libraries, IProgressMonitor monitor, boolean addToClasspath)
+          throws JavaModelException, CoreException {
     SubMonitor subMonitor = SubMonitor.convert(monitor,
         Messages.getString("adding.app.engine.libraries"), libraries.size()); //$NON-NLS-1$
 
     IClasspathEntry[] rawClasspath = javaProject.getRawClasspath();
-    List<IClasspathEntry> newRawClasspath = new ArrayList<>(rawClasspath.length + libraries.size());
+    
+    // + 1 because we pass the submonitor along below
+    List<IClasspathEntry> newRawClasspath = new ArrayList<>(rawClasspath.length + libraries.size() + 1);
```

Comment:
I don't really understand the +1. It seems that the size of `newRawClasspath` can never go over the size sum of the existing classpath entries (`rawClasspath`) and the libraries we attempt to add (`libraries`).

Personally, I don't care about optimization in this case and tend to just do `new ArrayList<>()`.